### ISSUE TITLE
`jsdoc/no-undefined-types` throws error for `globalThis[keyName]` (and maybe other types)

### ISSUE THREAD

**Issue by what1s1ove** [gajus/eslint-plugin-jsdoc#1566]
Hey guys! First of all, thanks for such an awesome plugin!

## Expected behavior
W/o the `jsdoc/no-undefined-types` error:

```js
class Storage {
	/** @type {globalThis.localStorage} */
	#storage
}
```

## Actual behavior
`The type 'globalThis.localStorage' is undefined jsdoc/no-undefined-types`

```js
class Storage {
	/** @type {globalThis.localStorage} */
	#storage
}
```

That resolves the issue, but of course, it would be nice to be able to use `globalThis`:
<img width="2762" height="336" alt="Image" src="https://github.com/user-attachments/assets/02416009-3889-4ed0-b6f9-45d58eec1c2f" />

## Environment

- Node version: 20.19.5
- ESLint version 9.37.0
- `eslint-plugin-jsdoc` version: 61.1.1
